app/ide-desktop/lib/dashboard/src/hooks/useLazyMemo.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * @file
+ *
+ * A hook that returns a memoized function that will only be called once
+ */
+
+import * as React from 'react'
+
+const UNSET_VALUE = Symbol('inset')
+
+/**
+ * A hook that returns a memoized function that will only be called once
+ */
+export function useLazyMemo<T>(factory: T | (() => T), deps: React.DependencyList): () => T {
+  return React.useMemo(() => {
+    let cachedValue: T | typeof UNSET_VALUE = UNSET_VALUE
+
+    return (): T => {
```

Comment:
the return type annotation is not strictly necessary here

Reply:
But also doesn't harm?

Reply:
true